ec/ecresource/projectresource/create_test.go

```diff
@@ -0,0 +1,302 @@
+				api.EXPECT().EnsureInitialised(ctx, createdModel).Return(nil)
+				api.EXPECT().Read(ctx, createdModel.Id.ValueString(), createdModel).Return(false, createdModel, nil)
+
+				handler := NewMockmodelHandler[resource_elasticsearch_project.ElasticsearchProjectModel](ctrl)
+				handler.EXPECT().ReadFrom(ctx, req.Plan).Return(&readModel, nil)
+				handler.EXPECT().GetID(createdModel).Return(createdModel.Id.ValueString()).AnyTimes()
+
+				return testData{
+					api:          api,
+					modelHandler: handler,
+					req:          req,
+					expectedDiags: diag.Diagnostics{
+						diag.NewErrorDiagnostic(
+							"Failed to read created elasticsearch project",
+							"The elasticsearch project was successfully created and initialised, but could then not be read back from the API",
+						),
+					},
+					expectedId: createdModel.Id.ValueStringPointer(),
+				}
+			},
+		},
+		{
+			name: "should set id in state, but ultimately fail if reading the initialised project returns an empty model",
+			testData: func(ctx context.Context) testData {
+				req := resource.CreateRequest{
+					Plan: tfsdk.Plan{
+						Raw: tftypes.NewValue(tftypes.Bool, true),
+					},
+				}
+
+				readModel := resource_elasticsearch_project.ElasticsearchProjectModel{
+					Name: basetypes.NewStringValue("name"),
+				}
+				createdModel := resource_elasticsearch_project.ElasticsearchProjectModel{
+					Id:   basetypes.NewStringValue("id"),
+					Name: basetypes.NewStringValue("name"),
+				}
+				finalModel := createdModel
+				finalModel.Id = basetypes.NewStringValue("final id")
+
+				api := NewMockapi[resource_elasticsearch_project.ElasticsearchProjectModel](ctrl)
+				api.EXPECT().Ready().Return(true)
+				api.EXPECT().Create(ctx, readModel).Return(createdModel, nil)
+				api.EXPECT().EnsureInitialised(ctx, createdModel).Return(nil)
+				api.EXPECT().Read(ctx, createdModel.Id.ValueString(), createdModel).Return(true, finalModel, nil)
+
+				handler := NewMockmodelHandler[resource_elasticsearch_project.ElasticsearchProjectModel](ctrl)
+				handler.EXPECT().ReadFrom(ctx, req.Plan).Return(&readModel, nil)
+				handler.EXPECT().GetID(createdModel).Return(createdModel.Id.ValueString()).AnyTimes()
+
+				return testData{
+					api:          api,
+					modelHandler: handler,
+					req:          req,
+					expectedId:   finalModel.Id.ValueStringPointer(),
+				}
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ctx := context.Background()
+			td := tt.testData(ctx)
+
+			res := resource.CreateResponse{
+				State: tfsdk.State{
+					Raw:    tftypes.NewValue(tftypes.Bool, true),
+					Schema: resource_elasticsearch_project.ElasticsearchProjectResourceSchema(ctx),
+				},
+			}
+
+			r := Resource[resource_elasticsearch_project.ElasticsearchProjectModel]{
+				api:          td.api,
+				modelHandler: td.modelHandler,
+				name:         "elasticsearch",
+			}
+
+			r.Create(ctx, td.req, &res)
```

Comment:
If `req` is removed, the line can be changed to `r.Create(ctx, resource.CreateRequest{}, &res)`